index/generator/library/library.go

```diff
@@ -705,6 +709,54 @@ func validateStackInfo(stackInfo schema.Schema, stackfolderDir string) []error {
 	return errors
 }
 
+// SetLastModifiedValue adds the last modified value to a pre-created index
+// The last modified dates are contained in a file named last_modified.json that is apart of the registry dir
+/* #nosec G304 -- lastModFile is produced from filepath.Join which cleans the input path */
+func SetLastModifiedValue(index []schema.Schema, registryDirPath string) ([]schema.Schema, error) {
+	lastModFile := filepath.Join(registryDirPath, "last_modified.json")
+	bytes, err := os.ReadFile(lastModFile)
+	if err != nil {
+		return index, err
+	}
+
+	var lastModifiedEntries schema.LastModifiedInfo
+	err = json.Unmarshal(bytes, &lastModifiedEntries)
+	if err != nil {
+		return index, err
+	}
+
+	lastModifiedEntriesMap := make(map[string]map[string]string)
+
+	for idx := range lastModifiedEntries.Stacks {
+		updateLastModifiedMap(lastModifiedEntriesMap, &lastModifiedEntries.Stacks[idx])
+	}
+
+	for idx := range lastModifiedEntries.Samples {
+		updateLastModifiedMap(lastModifiedEntriesMap, &lastModifiedEntries.Samples[idx])
+	}
+
+	for i := range index {
+		for j := range index[i].Versions {
+			updateSchemaLastModified(&index[i], j, lastModifiedEntriesMap[index[i].Name][index[i].Versions[j].Version])
```

Comment:
I'm wondering here if it's better to assign the params passed in this function to variables in order the code to be more readable. WDYT?